packages/jupyter-chat/src/components/chat-input.tsx

```diff
@@ -130,16 +130,22 @@ export function ChatInput(props: ChatInput.IProps): JSX.Element {
       return;
     }
 
-    // do not send the message if the user was selecting a suggested command from the
+    // Do not send the message if the user was selecting a suggested command from the
     // Autocomplete component.
     if (highlighted) {
       return;
     }
 
+    // Do not send empty messages, and avoid adding new line in empty message.
+    if (!input.trim()) {
+      event.stopPropagation();
+      event.preventDefault();
+      return;
+    }
```

Comment:
This handler only sends the message if `!input.trim()`, but the `SendButton` component is enabled by a different condition (when `input.length >= 0`).

```tsx
<SendButton
  model={model}
  sendWithShiftEnter={sendWithShiftEnter}
  inputExists={input.length > 0}
  onSend={onSend}
  hideIncludeSelection={hideIncludeSelection}
  hasButtonOnLeft={!!props.onCancel}
/>

```

Currently, if the input contains only white space, the send button will be enabled (which shouldn't happen), but pressing Enter/Shift-Enter does nothing. This is a difference in behavior caused by using two separate state variables.

I recommend defining a new local variable `const inputExists = !!input.trim()` in the component's top scope, and using that variable in both the handler & the send button.

Reply:
Good catch, thanks

Reply:
Updated in https://github.com/jupyterlab/jupyter-chat/pull/126/commits/b28a6983207a22119ef03ca4f5ea251ef62c1252